tests/test_constrain_current.py

```diff
@@ -2,118 +2,190 @@
 
 import numpy as np
 import pytest
+from tests.test_axis_limits import not_implemented_limits
 
 import desc.io
 from desc.compute import compute as compute_fun
-from desc.compute import get_params, get_profiles, get_transforms
+from desc.compute import data_index, get_params, get_profiles, get_transforms
 from desc.grid import ConcentricGrid, LinearGrid, QuadratureGrid
 
 
 class _ExactValueProfile:
     """Monkey patches the compute method of desc.Profile for testing."""
 
-    def __init__(self, eq, grid):
-        self.eq = eq
-        self.grid = grid
-        self.transforms = get_transforms("current_rr", eq=eq, grid=grid)
-        self.profiles = get_profiles("current_rr", eq=eq, grid=grid)
-        self.params = get_params("current_rr", eq=eq, has_axis=grid.axis.size)
+    def __init__(self, eq, grid, profile_name):
+        names = []
+        dr = 0
+        while profile_name + "_" * bool(dr) + "r" * dr in data_index:
+            names += [profile_name + "_" * bool(dr) + "r" * dr]
+            dr += 1
+        data = compute_fun(
+            names=names,
+            params=get_params(keys=names, eq=eq, has_axis=grid.axis.size),
+            transforms=get_transforms(keys=names, eq=eq, grid=grid),
+            profiles=get_profiles(keys=names, eq=eq, grid=grid),
+        )
+        self.output = {dr: data[name] for dr, name in enumerate(names)}
 
     def compute(self, params, dr, *args, **kwargs):
-        if dr == 0:
-            # returns the surface average of B_theta in amperes
-            return compute_fun(
-                "current",
-                params=self.params,
-                transforms=self.transforms,
-                profiles=self.profiles,
-            )["current"]
-        if dr == 1:
-            # returns the surface average of B_theta_r in amperes
-            return compute_fun(
-                "current_r",
-                params=self.params,
-                transforms=self.transforms,
-                profiles=self.profiles,
-            )["current_r"]
-        if dr == 2:
-            # returns the surface average of B_theta_rr in amperes
-            return compute_fun(
-                "current_rr",
-                params=self.params,
-                transforms=self.transforms,
-                profiles=self.profiles,
-            )["current_rr"]
+        return self.output.get(dr, np.nan)
 
 
 class TestConstrainCurrent:
-    """Tests for running DESC with a fixed current profile."""
+    """Tests for computing iota from fixed current profile and vice versa.
 
-    @pytest.mark.unit
+    The equations referenced in the descriptions below refer to the document
+    attached to the description of GitHub pull request #556.
+    """
+
+    @pytest.mark.regression
     @pytest.mark.solve
-    def test_compute_rotational_transform(self, DSHAPE, HELIOTRON_vac):
```

Comment:
This test with `HELIOTRON_vac` has been useless (wasn't testing anything) since we changed how vacuum equilibrium were solved in September. Moving it to the new test below resolves this issue.

Reply:
responded here https://github.com/PlasmaControl/DESC/pull/556#discussion_r1272669208